src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -681,10 +680,8 @@ private unsafe void RemoveNodeFromDatabase(TransactionOperationContext context,
             using (Slice.From(context.Allocator, "db/" + databaseName, out Slice key))
             {
                 if (items.ReadByKey(lowerKey, out TableValueReader reader) == false)
-                {
-                    NotifyLeaderAboutError(index, leader, new InvalidOperationException($"The database {databaseName} does not exists"));
-                    return;
-                }
+                    throw new RachisApplyException($"The database {databaseName} does not exists");
```

Comment:
I wonder if we can get to a state where the database doesn't exist since we haven't created it yet on this node and if so this is a case that could be specific to some nodes and not consistent across the cluster.
I know that if there is a creation command then it should be executed before the deletion but we have a two-phase deletion process and I have a gut feeling that there is an edge case lurking here.

Reply:
No, because this is running in the cluster, so this is running in the same sequence everywhere